src/utilities.js

```diff
@@ -31,6 +35,8 @@ export var $extend = function (destination) {
       if (source[property] && $isplainobject(source[property])) {
         if (!destination.hasOwnProperty(property)) destination[property] = {}
         $extend(destination[property], source[property])
+      } else if (Array.isArray(source[property])) {
+        destination[property] = deepcopy(source[property])
       } else {
```

Comment:
the Array in `$extend ` is always extended by `deepcopy`.

Reply:
So we can close #506 when this is merged?

Reply:
> So we can close #506 when this is merged?

In my opinion, the deepcopy of array is unnecessary, however a copy of an `Array` itself(by `destination[property] = source[property].concat()`) will be better, it cost less memory.